### ISSUE TITLE
[Proposal] From routes, allow to call middleware as method

### ISSUE THREAD

**Issue by epalmans** [laravel/ideas#1544]
instead of

```php
Route::get('/{company}/edit', 'CompanyController@index')->middleware('can:update,company');
```
also allow it to written as:
```php
Route::get('/{company}/edit', 'CompanyController@index')->can('update', 'company');
// OR
Route::get('/{company}/edit', 'CompanyController@index')->can('update,company');
```

------

Sample Implementation:
```php
// in laravel\framework\src\Illuminate\Routing\Route.php

public function __call($method, $args)
{
	$this->middleware($args ? "{$method}:" . implode($args, ",") : $method);

	return $this;
}

// TODO (maybe?): check if middleware (= $method) exists
```

**Comment by svenluijten** [comment]
What if you alias your as middleware `'group'`?

**Comment by epalmans** [comment]
> What if you alias your middleware as `'group'`?

With the proposal you could use the aliases for routemiddleware groups too; for instance:

Route::get('/{company}/edit', 'CompanyController@edit')->api();

**Comment by svenluijten** [comment]
I realize now that my question actually made very little sense, sorry about that.

What I meant was; what if you create an alias for your middleware, and name it the same as another method already on the `Route` object (like `name`)? With your proposed change, that will end up looking like this:

```php
Route::get(...)->name();
```

But obviously the effect won't be what you'd expect as a user. After this change would be implemented, extra care would have to be put into adding methods to the `Route` class, seeing as some applications might break if a new method is added that's already someone's middleware alias.

I think "opening the flood gates" like this would cause far more trouble than it's worth, and something that'd allow you to do `->middleware('can', 'update', 'company');` or `->middleware('can', ['update', 'company']);` would be more readable anyway.

**Comment by epalmans** [comment]
that's correct. Methods with the same name already existing on the Route class would be called instead. The proposal would just serve as a 'fluent' alternative to call middleware. Definately not intended as a replacement to existing means. Also, because middleware aliases might contain characters that aren't allowed in functionnames
